jf_agent/agent_logging.py

```diff
@@ -72,3 +123,15 @@ def log_and_print(logger, level, msg, exc_info=False):
     print(msg, flush=True)
     if exc_info:
         print(traceback.format_exc())
+
+
+def log_and_print_error_or_warning(logger, level, error_code,msg_args=[], exc_info=False):
+    '''
+    For a failure that should be sent to the logger, and also written
+    to stdout (for user visibility)
```

Comment:
You could make the docstring for `log_and_print` (rather than `log_and_print_error_or_warning`) more accurate in the sense that it's no longer "For a failure", it's now more "For an info-level message" or something.